src/java.base/share/classes/jdk/internal/classfile/impl/StackMapDecoder.java

```diff
@@ -131,14 +165,25 @@ public int compare(final StackMapFrameInfo o1, final StackMapFrameInfo o2) {
             if (offset == prevOffset) {
                 throw new IllegalArgumentException("Duplicated stack frame bytecode index: " + offset);
             }
-            writeFrame(buf, offset - prevOffset - 1, prevLocals, fr);
+            writeFrame(buf, offset - prevOffset - 1, prevLocals, prevUnsets, fr);
             prevOffset = offset;
             prevLocals = fr.locals();
+            prevUnsets = fr.unsetFields();
         }
     }
 
-    private static void writeFrame(BufWriterImpl out, int offsetDelta, List<VerificationTypeInfo> prevLocals, StackMapFrameInfo fr) {
+    private static void writeFrame(BufWriterImpl out, int offsetDelta, List<VerificationTypeInfo> prevLocals, List<NameAndTypeEntry> prevUnsets, StackMapFrameInfo fr) {
         if (offsetDelta < 0) throw new IllegalArgumentException("Invalid stack map frames order");
+        // enclosing frames
+        writeLarvalFrame:
```

Comment:
not a big fan of using labels, I wonder if the code can be refactored to avoid using them

Reply:
Yep, factored these out to methods with extra asserts, does look cleaner